src/test/kotlin/com/cultureamp/kafka/connect/plugins/transforms/ClickHouseFlattenTransformerTest.kt

```diff
@@ -495,4 +368,43 @@ class ClickHouseFlattenTransformerTest {
             "com.mongodb.kafka.connect.source.json.formatter.DefaultJson",
             JsonWriterSettingsProvider::class.java
         ).jsonWriterSettings
+
+    /**
+     * Helper function to compare Struct values field by field.
+     * Needed because Struct.equals() compares object identity, not content.
```

Comment:
Not from what I can see
```
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Struct struct = (Struct) o;
        return Objects.equals(schema, struct.schema) &&
                Arrays.deepEquals(values, struct.values);
    }
```

But it does compare the schema, perhaps that's the problem?